### PR TITLE
add System.Diagnostics.DiagnosticSource 9.0.0

### PR DESCRIPTION
**Pull request by JanProvaznik**
Needed for MSBuild telemetry investments. https://github.com/dotnet/msbuild/issues/11169

1. ./generate.sh --package System.Diagnostics.DiagnosticSource,9.0.0
2. add DependencyPackageProjects to eng/Build.props

### CODE REVIEW

File: eng/SourceBuildPrebuiltBaseline.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <!-- runtime prebuilts.  Update when runtime produces intermediate packages correctly. -->
     <UsagePattern IdentityGlob="runtime.linux-x64.Microsoft.NETCore.ILAsm/*" />
     <UsagePattern IdentityGlob="runtime.linux-x64.Microsoft.NETCore.ILDAsm/*" />
+    <UsagePattern IdentityGlob="Microsoft.NETCore.App.Ref/*9.0.0*" />
```
Comment (MichaelSimons) [inline eng/SourceBuildPrebuiltBaseline.xml:9]: This can't be added.  I think I know how to fix this and will push a fix.

**Comment by JanProvaznik** [comment]
@dotnet/source-build I did quite weird stuff to get this PR green, please review how it should be done correctly.

1. it failed upon seeing the `scoped` keywords in the generated files https://github.com/dotnet/source-build-reference-packages/pull/1101/checks?check_run_id=34712767234 fixed by: [`c48afa6` (#1101)](https://github.com/dotnet/source-build-reference-packages/pull/1101/commits/c48afa6e0c362045562363e1429d84f2b29b7916)  [`c162c9b` (#1101)](https://github.com/dotnet/source-build-reference-packages/pull/1101/commits/c162c9b22c75a3c648c61f12818ff4f8067f3589)
2. It failed with `error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported` https://github.com/dotnet/source-build-reference-packages/pull/1101/checks?check_run_id=34714580836 resolved by [`c162c9b` (#1101)](https://github.com/dotnet/source-build-reference-packages/pull/1101/commits/c162c9b22c75a3c648c61f12818ff4f8067f3589)
3. a prebuilt was reported for microsoft.netcore.app.ref which in fact exists in the repo https://github.com/dotnet/source-build-reference-packages/tree/main/src/targetPacks/ILsrc/microsoft.netcore.app.ref , resolved by [`40967d8` (#1101)](https://github.com/dotnet/source-build-reference-packages/pull/1101/commits/40967d85ff8627f44e6534fa0bf9bbf87a18aac5)

**Comment by MichaelSimons** [comment]
> 3. a prebuilt was reported for microsoft.netcore.app.ref which in fact exists in the repo https://github.com/dotnet/source-build-reference-packages/tree/main/src/targetPacks/ILsrc/microsoft.netcore.app.ref , resolved by [`40967d8` (#1101)](https://github.com/dotnet/source-build-reference-packages/pull/1101/commits/40967d85ff8627f44e6534fa0bf9bbf87a18aac5)

I think I took care of this in https://github.com/dotnet/source-build-reference-packages/pull/1101/commits/1de36a7e7affbdb8ae03073a17d14c8b9924ce60

**Review by ellahathaway — APPROVED** [review]

**Comment by MichaelSimons** [comment]
>It failed with error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported https://github.com/dotnet/source-build-reference-packages/pull/1101/checks?check_run_id=34714580836 resolved by [c162c9b (#1101)](https://github.com/dotnet/source-build-reference-packages/pull/1101/commits/c162c9b22c75a3c648c61f12818ff4f8067f3589)

This is a known problem tracked with https://github.com/dotnet/source-build/issues/4159.  We have a common shared file with the type in it.  I update the project to reference it.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**